### PR TITLE
MISC minidump-stackwalk CLI work

### PR DESCRIPTION
**Pull request by Gankra**
See individual commits for details.

### CODE REVIEW

**Comment by Gankra** [comment]
Oh, fixes #296

File: minidump-processor/src/process_state.rs [hunk 1]
```diff
@@ -389,6 +389,17 @@ impl ProcessState {
     /// This is very verbose, it implements the output format used by
     /// minidump_stackwalk.
     pub fn print<T: Write>(&self, f: &mut T) -> io::Result<()> {
+        self.print_internal(f, false)
+    }
+
+    /// Write a brief human-readable description of the process state to `f`.
```
Comment (luser) [inline minidump-processor/src/process_state.rs:395]: Ooh, I like this!

File: minidump-stackwalk/src/main.rs [hunk 2]
```diff
@@ -16,56 +16,108 @@ use minidump_processor::{
...
         .setting(AppSettings::NextLineHelp)
         .setting(AppSettings::ColoredHelp)
+        .setting(AppSettings::DeriveDisplayOrder)
+        .usage("minidump-stackwalk [FLAGS] [OPTIONS] <minidump> [--] [symbols-path]...")
+        .arg(
+            Arg::with_name("json")
+                .long("json")
+                .long_help("Emit a machine-readable JSON report (the default).
+
+Official schema TBD, but the output *is* stabilized. That said, sometimes values can't be computed \
```
Comment (luser) [inline minidump-stackwalk/src/main.rs:39]: FYI the original implementation bug [had a quasi-schema attached](https://bug573100.bmoattachments.org/attachment.cgi?id=549163). It's assuredly out of date and not an actual JSON schema in any event.

Comment (Gankra) [reply minidump-stackwalk/src/main.rs:39]: yeah process_state's comments are nearly a complete schema already, I just need to pull them out to a proper location and add some more notes.

**Pull request merged**